src/plugins/console/public/application/models/legacy_core_editor/mode/input_highlight_rules.js

```diff
@@ -88,10 +88,36 @@ export function InputHighlightRules() {
       addEOL(['url.param'], /([^&=]+)/, 'start-sql'),
       addEOL(['url.amp'], /(&)/, 'start-sql')
     ),
+    comments: [
+      {
+        token: ['comment.punctuation', 'comment.line'],
+        regex: /(#)(.*$)/,
+      },
+      {
+        token: 'comment.punctuation',
+        regex: /\/\*/,
+        push: [
+          {
+            token: 'comment.punctuation',
+            regex: /\*\//,
+            next: 'pop',
+          },
+          {
+            defaultToken: 'comment.block',
+          },
+        ],
+      },
+      {
+        token: ['comment.punctuation', 'comment.line'],
+        regex: /(\/\/)(.*$)/,
+      },
+    ],
   };
 
   addXJsonToRules(this);
 
+  this.$rules.json.unshift({ include: 'comments' });
```

Comment:
nit: what if we move that logic to addXJsonToRules. With passing a configuration option for that

Reply:
It requires modifying the `addXJsonToRules` method in `kbn/ace` package. Also, we might need to duplicate the rules for comments in Console's input highlight rules and in `kbn/ace`.